device-connectors/src/testflinger_device_connectors/devices/zapper_iot/__init__.py

```diff
@@ -37,16 +37,41 @@ def _validate_configuration(
         Validate the job config and data and prepare the arguments
         for the Zapper `provision` API.
         """
-        try:
-            provision_plan = self.job_data["provision_data"]["provision_plan"]
-            validate_provision_plan(provision_plan)
-        except KeyError as e:
-            raise ProvisioningError from e
 
+        username = self.job_data.get("test_data", {}).get(
+            "test_username", "ubuntu"
+        )
+        password = self.job_data.get("test_data", {}).get(
+            "test_password", "ubuntu"
+        )
+
+        provisioning_data = {
+            "username": username,
+            "password": password,
+            "preset": self.job_data["provision_data"].get("preset"),
```

Comment:
So the "preset" is the same level as urls and provision_plan right?

Reply:
yes it is, same level, and optional